### PR TITLE
Adopt Christina Yang's (experimental) tui

### PR DESCRIPTION
**Pull request by Bluesy1**
As per the title.

Installable via `pip install "problem_bank_scripts[tui] @ git+https://github.com/open-resources/problem_bank_scripts@adopt-yangchristina-tui"` currently.

After installation, the tui should be usable via `pbs-cli create-question .....`, or `python -m problem_bank_scripts.tui ...`

If testing via the command above, I recommend using a fresh venv (via, for e.g., `python -m venv ...`)

More details about the tui can be found at https://github.com/yangchristina/opb-utils/blob/main/tui/README.md

Example usage:

```console
gavin@wsl:~$ pbs-cli --help
usage: pbs-cli [-h] {checkq,check-server-ast,lint-server,process-q,process,create-question} ...

OPB Problem Bank Scripts CLI

positional arguments:
  {checkq,check-server-ast,lint-server,process-q,process,create-question}
    checkq              Check the syntax of a single question.
    check-server-ast    Check server code ast for one or more markdown files.
    lint-server         Lint the server code ast for one or more markdown files.
    process-q           Check the syntax of a single question.
    process             Processes all questions.
    create-question     Create a barebones OPB markdown question via a series of tui prompts.

options:
  -h, --help            show this help message and exit
gavin@wsl:~$ pbs-cli create-question --help
usage: pbs-cli create-question [-h] [--create-pr | --no-create-pr] [--gpt] [--saved-json SAVED_JSON]
                               [--env-file ENV_FILE | --no-env-file]

Create a barebones OPB markdown question via a series of tui prompts. It requires the following environment variables to
be set, either via '--env-file', or in the shell's environment: 'GITHUB_USERNAME', 'WRITE_PATH', 'PL_QUESTION_PATH',
'MY_NAME', and 'MY_INITIALS'

options:
  -h, --help                   show this help message and exit
  --create-pr, --no-create-pr  Create a PR with the generated question.
  --gpt                        Use ChatGPT gpt-4o-mini to generate the MCQ options and number input code. If this is
                               specified, the environment variable 'OPENAI_API_KEY' is also required to be available, or
                               be in the env file specified by --env-file. (default: False)
  --saved-json SAVED_JSON      File to read/save tui prompts to/from. (default: ./saved.json)
  --env-file ENV_FILE          Path to the environment file with the required environment variables (default: .env)
  --no-env-file                Do not read from any environment file. If you use this, make sure the required
                               environment variables are set in your shell's environment. (default: False)

This program requires both 'git' and 'gh' to be installed to create PRs. You can find the installation instructions for
'gh' at https://github.com/cli/cli#installation.
```

### CODE REVIEW

**Comment by Bluesy1** [comment]
cc @yangchristina

File: src/problem_bank_scripts/tui/tui.py [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+            variables = exercise["variables"]
+    try:
+        set_default(exercise, key="assets", value=[], saved=saved)
+
+        textbook = exercise.get("attribution", None)
+        if textbook is None:
+            textbook = questionary.select(
+                message="What textbook is this from? (Select standard if not from any on the list)",
+                choices=KNOWN_ATTRIBUTIONS,
+            ).ask()
```
Comment (yangchristina) [inline src/problem_bank_scripts/tui/tui.py:251]: Actually can the default textbook be set with a .env variable?

```suggestion
            textbook = os.environ["TEXTBOOK"] if "TEXTBOOK" in os.environ else questionary.select(
                message="What textbook is this from? (Select standard if not from any on the list)",
                choices=KNOWN_ATTRIBUTIONS,
            ).ask()
```

Comment (Bluesy1) [reply src/problem_bank_scripts/tui/tui.py:251]:
```suggestion
            textbook = os.environ.get("TEXTBOOK") or questionary.select(
                message="What textbook is this from? (Select standard if not from any on the list)",
                choices=KNOWN_ATTRIBUTIONS,
            ).ask()
```

**Comment by Bluesy1** [comment]
Docs/instructions now available at https://problem-bank-scripts.readthedocs.io/en/latest/question_creator.html

Use `pip install "problem_bank_scripts[tui] @ git+https://github.com/open-resources/problem_bank_scripts" to install now, instead of the command in the pr description

**Pull request merged**